src/index.jsx

```diff
@@ -1,79 +1,21 @@
-import { Router, Route, browserHistory, IndexRoute } from 'react-router';
-import { syncHistoryWithStore, routerReducer as routing, routerMiddleware } from 'react-router-redux';
 import injectTapEventPlugin from 'react-tap-event-plugin';
-import appReducer, { REDUCER_KEY } from 'reducers';
+import RouterWithProvider from 'components/Router';
+import store from 'store';
 import { getMetadata } from 'actions';
-import App from 'components/App';
-import Match from 'components/Match';
-import Player from 'components/Player';
-import Home from 'components/Home';
-import Search from 'components/Search';
-import Explorer from 'components/Explorer';
-import FourOhFour from 'components/FourOhFour';
-import { Heroes } from 'components/Heroes';
-import Request from 'components/Request';
-import Distributions from 'components/Distributions';
-// Load CSS
+
+// Import css for webpack to load
 import 'c3/c3.css';
 import './index.css';
-// require('./node_modules/dota2-minimap-hero-sprites/assets/stylesheets/dota2minimapheroes.css');
 
-// Promise polyfill for IE
+// Promise polyfill for IE used by fetch
 require('es6-promise').polyfill();
-
 // This is used by material-ui components
 injectTapEventPlugin();
 
-const loggerMiddleware = createLogger();
-
-const reducer = combineReducers({
-  [REDUCER_KEY]: appReducer,
-  routing,
-});
-const store = createStore(reducer, compose(
-  applyMiddleware(thunkMiddleware),
-  applyMiddleware(loggerMiddleware),
-  applyMiddleware(routerMiddleware(browserHistory)),
-  window.devToolsExtension ? window.devToolsExtension() : f => f // This enables the redux dev tools extension, or does nothing if not installed
-));
-
 // Fetch metadata (used on all pages)
 store.dispatch(getMetadata());
-// Create an enhanced history that syncs navigation events with the store
-const history = syncHistoryWithStore(browserHistory, store);
 
 const reactElement = document.getElementById('react');
-render(
-  <Provider store={store}>
-    {/* Tell the Router to use our enhanced history */}
-    <Router history={history}>
-      <Route path="/" component={App}>
-        <IndexRoute component={Home} />
-        <Route path="search" component={Search} />
-        <Route path="matches/:match_id" component={Match}>
-          <Route path=":info" />
-        </Route>
-        <Route path="players/:accountId" component={Player}>
-          <Route path=":info">
-            <Route path=":subInfo" />
-          </Route>
-        </Route>
-        <Route path="explorer" component={Explorer} />
-        <Route path="heroes(/:hero_id)" component={Heroes}>
-          <Route path=":info" />
-        </Route>
-        <Route path="request" component={Request} />
-        <Route path="distributions" component={Distributions} />
-        <Route path="carry" />
-        <Route path="status" />
-        <Route path="*" component={FourOhFour} />
-      </Route>
-    </Router>
-  </Provider>, reactElement);
+render(<RouterWithProvider />, reactElement);
```

Comment:
maybe pass the store to the RouterWithProvider instead of re-importing it there since you already need a store reference here anyway?